src/test_lib_json/fuzz.cpp

```diff
@@ -3,47 +3,31 @@
 // recognized in your jurisdiction.
 // See file LICENSE for detail or copy at http://jsoncpp.sourceforge.net/LICENSE
 
-#include "fuzz.h"
-
-#include <cstdint>
-#include <json/config.h>
-#include <json/json.h>
-#include <memory>
-#include <stdint.h>
+#include <sstream>
 #include <string>
-
-namespace Json {
-class Exception;
-}
-
-extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
-  Json::CharReaderBuilder builder;
-
-  if (size < sizeof(uint32_t)) {
+#include "fuzz.h"
+#include "json/config.h"
+#include "json/features.h"
+#include "json/reader.h"
+#include "json/value.h"
+#include "json/writer.h"
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+  std::string json_string(reinterpret_cast<const char *>(data), size);
+  Json::Reader reader(Json::Features::strictMode());
+  Json::Value value;
+  const bool success = reader.parse(json_string, value, false);
+  if (!success) {
     return 0;
   }
 
-  uint32_t hash_settings = *(const uint32_t*)data;
-  data += sizeof(uint32_t);
-
-  builder.settings_["failIfExtra"] = hash_settings & (1 << 0);
-  builder.settings_["allowComments_"] = hash_settings & (1 << 1);
-  builder.settings_["strictRoot_"] = hash_settings & (1 << 2);
-  builder.settings_["allowDroppedNullPlaceholders_"] = hash_settings & (1 << 3);
-  builder.settings_["allowNumericKeys_"] = hash_settings & (1 << 4);
-  builder.settings_["allowSingleQuotes_"] = hash_settings & (1 << 5);
-  builder.settings_["failIfExtra_"] = hash_settings & (1 << 6);
-  builder.settings_["rejectDupKeys_"] = hash_settings & (1 << 7);
-  builder.settings_["allowSpecialFloats_"] = hash_settings & (1 << 8);
+  // Write with StyledWriter
+  Json::StyledWriter styled_writer;
```

Comment:
Should we use a method that is not deprecated?

Reply:
Feel free to add an other fuzzer for the new method.
I think it's worthwhile using the *deprecated* one, because people are still using it.